docs/src/test-sharding-js.md

```diff
@@ -0,0 +1,169 @@
+        shard: [1/4, 2/4, 3/4, 4/4]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 18
+    - name: Install dependencies
+      run: npm ci
+    - name: Install Playwright browsers
+      run: npx playwright install
+
+    - name: Run Playwright tests
+      run: npx playwright test --shard ${{ matrix.shard }}
+```
+
+## Creating combined report in GitHub Actions
+
+Each test shard will have its own test report. If you want to have a combined report showing all tests results from all shards you need to add a job that will merge individual shard reports.
+
+### Configuring shard reporting
+
+We start by adding `blob` reporter to the config:
+
+```js
+export default defineConfig({
+  testDir: './tests',
+  reporter: [['blob', { outputDir: 'blob-report' }]],
+});
+```
+
+Blob report contains information about all the tests that were run and their results as well as all test attachments such as traces and screenshot diffs. Blob reports can be merged and converted to any other Playwright report.
+
+### Uploading shard report
+
+To merge individual reports they need to be copied into a shared location. GitHub Actions Artifacts is a convenient mechanism that lets you do that. By adding the following step after the test execution we upload blob report from each shard into GitHub Actions Artifact with name`blob-report-${{ github.run_attempt }}` (this is essentially a shared directory where each shard will copy its report to):
+
+```yaml
+jobs:
+  test:
+...
+    - name: Upload blob report to Artifacts
+      if: always()
+      uses: actions/upload-artifact@v3
+      with:
+        name: blob-report-${{ github.run_attempt }}
+        path: blob-report
+        retention-days: 2
+```
+
+### Merge reports job
+
+After all shards finished running it's time to run a job that will merge the reports and produce a combined HTML report. To ensure the execution order, we make `merge-report` job [depend](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#defining-prerequisite-jobs) on our sharded `test` job. `merge-report` job below reads all blob reports from `blob-report-${{ github.run_attempt }}` artifact and produces a single HTML report.
+
+```yaml
+jobs:
+  merge-report:
+    if: always()
+    needs: [test]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 18
+    - name: Install dependencies
+      run: npm ci
+
+    - name: Download Blob Reports from Artifacts
+      uses: actions/download-artifact@v3
+      with:
+        name: blob-report-${{ github.run_attempt }}
+        path: blob-report
+
+    - name: Merge into HTML Report
+      run: |
+        npx playwright merge-reports ./blob-report --reporter html
+
+    - name: Upload HTML report
+      uses: actions/upload-artifact@v3
+      with:
+        name: html-report-${{ github.run_attempt }}
```

Comment:
I think we can drop run attempt everywhere, since its already per run afaik.

Reply:
I'm not sure about this, let's discuss in the meeting.